src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Raw.cs

```diff
@@ -141,15 +141,15 @@ public void WriteRawValue(ReadOnlySequence<byte> utf8Json, bool skipInputValidat
                 ValidateWritingValue();
             }
 
-            long len = utf8Json.Length;
+            int len = checked((int)utf8Json.Length);
```

Comment:
The cast should be moved after the check so that large inputs always always surface as the same exception (`ArgumentException` rather than `OverflowException`).

Reply:
This single commit change was just because of [comment](https://github.com/dotnet/runtime/pull/76444#discussion_r987853288), checked cast seems prior art for ROSeq.Length in this space.

Reply:
In any case, we are throwing two different exceptions for what is essentially the same failure mode. I would recommend making it just one (and preferably ArgumentException since we can control the error message, unlike checked casts).

Reply:
Fixed.